server/polar/organization/schemas.py

```diff
@@ -86,33 +86,24 @@ class OrganizationCreate(OrganizationPrivateBase):
     pledge_minimum_amount: int = settings.MINIMUM_ORG_PLEDGE_AMOUNT
 
     @classmethod
-    def from_github_installation(
-        cls, installation: github.rest.Installation
-    ) -> OrganizationCreate:
-        account = installation.account
-
-        if not isinstance(account, github.rest.SimpleUser):
-            raise Exception("Polar does not support GitHub Enterprise")
-
-        is_personal = account.type.lower() == "user"
-        name = account.login
-        avatar_url = account.avatar_url
-        external_id = account.id
-        if not name:
-            raise Exception("repository.name is not set")
-        if not avatar_url:
-            raise Exception("repository.avatar_url is not set")
-
+    def from_github(
+        cls,
+        user: github.webhooks.User | github.rest.SimpleUser,
+        *,
+        installation: github.rest.Installation | None = None,
+    ) -> Self:
         return cls(
             platform=Platforms.github,
-            name=name,
-            external_id=external_id,
-            avatar_url=avatar_url,
-            is_personal=is_personal,
-            installation_id=installation.id,
-            installation_created_at=installation.created_at,
-            installation_updated_at=installation.updated_at,
-            installation_suspended_at=installation.suspended_at,
+            name=user.login,
+            external_id=user.id,
+            avatar_url=user.avatar_url,
+            is_personal=user.type.lower() == "user",
+            installation_id=installation.id if installation else None,
+            installation_created_at=installation.created_at if installation else None,
+            installation_updated_at=installation.updated_at if installation else None,
+            installation_suspended_at=installation.suspended_at
+            if installation
+            else None,
```

Comment:
Is there a risk that we're now unsetting the `installation_` and other fields that default to null in the database?

Could we have a separate model for this other use case, and remove the fields from this model?

Like this maybe? **(incomplete example)**

```python
class OrganizationCreate(OrganizationPrivateBase):
    pledge_minimum_amount: int = settings.MINIMUM_ORG_PLEDGE_AMOUNT

    @classmethod
    def from_github(
        cls,
        user: github.webhooks.User | github.rest.SimpleUser,
    ) -> Self:
        return cls(
            platform=Platforms.github,
            name=user.login,
            external_id=user.id,
            avatar_url=user.avatar_url,
            is_personal=user.type.lower() == "user",
        )


class OrganizationInstallationUpdate(OrganizationCreate):
    installation_id: int
    installation_created_at: datetime
    installation_updated_at: datetime | None = None
    installation_suspended_at: datetime | None = None

    @classmethod
    def from_github(
        cls,
        user: github.webhooks.User | github.rest.SimpleUser,
        installation: github.rest.Installation,
    ) -> Self:
        return cls(
            platform=Platforms.github,
            name=user.login,
            external_id=user.id,
            avatar_url=user.avatar_url,
            is_personal=user.type.lower() == "user",
            installation_id=installation.id,
            installation_created_at=installation.created_at,
            installation_updated_at=installation.updated_at,
            installation_suspended_at=installation.suspended_at
            if installation
            else None,
        )
````

Reply:
In general, my strategy has been to avoid schemas with nullable fields, and instead create multiple schemas. :)

Reply:
You're right, this would have reset `installation_` fields.

I've taken back the approach with `exclude_unset`. I really think it's a good way of managing this kind of cases. Having several separate schemas would have been hard to handle with the `.update` method.